test/core/iomgr/endpoint_pair_test.cc

```diff
@@ -18,18 +18,108 @@
 
 #include "src/core/lib/iomgr/endpoint_pair.h"
 
+#include <chrono>
+
 #include <gtest/gtest.h>
 
+#include <grpc/event_engine/event_engine.h>
 #include <grpc/grpc.h>
 #include <grpc/support/alloc.h>
 #include <grpc/support/log.h>
 #include <grpc/support/time.h>
 
+#include "src/core/lib/address_utils/parse_address.h"
+#include "src/core/lib/channel/channel_args.h"
+#include "src/core/lib/event_engine/channel_args_endpoint_config.h"
+#include "src/core/lib/event_engine/default_event_engine.h"
+#include "src/core/lib/experiments/experiments.h"
 #include "src/core/lib/gpr/useful.h"
 #include "src/core/lib/gprpp/crash.h"
+#include "src/core/lib/gprpp/notification.h"
+#include "src/core/lib/iomgr/event_engine_shims/endpoint.h"
+#include "src/core/lib/resource_quota/memory_quota.h"
 #include "test/core/iomgr/endpoint_tests.h"
+#include "test/core/util/port.h"
 #include "test/core/util/test_config.h"
 
+#ifdef GRPC_POSIX_SOCKET_TCP
+
+using namespace std::chrono_literals;
+
+namespace {
+
+using ::grpc_event_engine::experimental::ChannelArgsEndpointConfig;
+using ::grpc_event_engine::experimental::EventEngine;
+
+EventEngine::ResolvedAddress URIToResolvedAddress(std::string address_str) {
```

Comment:
This should probably live next to `ResolvedAddressToURI` in `src/core/lib/event_engine/tcp_socket_utils.h`, but return `absl::StatusOr<ResolvedAddress>`

Reply:
Moved it to src/core/lib/event_engine/tcp_socket_utils.h. Also had to create a new target called parse_address in BUILD. This contains the functions necessary to implement the URIToResolvedAddress Method.